src/blockchain_db/sqlite/db_sqlite.cpp

```diff
@@ -198,39 +195,45 @@ std::optional<std::vector<cryptonote::batch_sn_payment>> BlockchainSQLite::get_s
   return payments;
 }
 
-std::vector<cryptonote::batch_sn_payment> BlockchainSQLite::calculate_rewards(const cryptonote::block& block, std::vector<cryptonote::batch_sn_payment> contributors)
+std::vector<cryptonote::batch_sn_payment> BlockchainSQLite::calculate_rewards(uint8_t hf_version, uint64_t distribution_amount, service_nodes::service_node_info sn_info)
 {
```

Comment:
This now takes in an amount to distribute and then the service_node_info which it uses to find the contributors, their percentages and also the operator fee